src/linter/linter.ts

```diff
@@ -87,16 +87,19 @@ export async function lintProject({
 export async function lintFile({
 	rootDir, pathsToLint, namespace, reportCoverage, includeMessageDetails,
 }: LinterOptions): Promise<LintResult[]> {
-	const reader = createReader({
-		fsBasePath: rootDir,
-		virBasePath: "/",
-	});
 	let resolvedFilePaths;
+	let virBasePath = "";
 	if (pathsToLint?.length) {
 		const absoluteFilePaths = resolveFilePaths(rootDir, pathsToLint);
 		resolvedFilePaths = transformFilePathsToVirtualPaths(
 			absoluteFilePaths, rootDir, "/", rootDir);
+		// Extract the (virtual) path from the filename
+		virBasePath = resolvedFilePaths[0].split("/").slice(0, -1).join("/");
```

Comment:
Hmmm, you're right about the deeply nested components 🤔 

But, how to extract the namespace without breaking the signature of the function?
Maybe, reuse the logic from `lintProject()`, but then we need a project, not a single file...

Reply:
I reverted the commit and all tests are still green 😅

Reply:
Please take a look here: [src/linter/linter.ts](https://github.com/SAP/ui5-linter/pull/73/files#diff-b5f6735c1e11d603b4261047a776b51e3fe83892f36e19a6a0506f59ab152032L92)

This is the original change compared to the main branch.
The change is working pretty well, except that it cannot handle `/test-resources/` and that's why I needed to find a better way of doing it.
Please take a look at the conversation we had with @flovogt on that matter here: https://github.com/SAP/ui5-linter/pull/73/files#r1601613818

Reply:
Got it, thanks. Since the `lintFile` API is currently only used in our tests, I think we can safely go with the current state and ignore the test-resources case for now 👍